go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -177,27 +79,15 @@ func (cm *ClusterManager) start(ctx context.Context) error {
 		}
 	}
 
-	cm.isActive = true
 	return nil
 }
 
-// stop managing the cluster operations by stopping the default
-// network cluster controller
-func (cm *ClusterManager) stop() error {
-	cm.Lock()
-	defer cm.Unlock()
-	if !cm.isActive {
-		// Is not active.  Nothing to halt.
-		return nil
-	}
-
+// Stop the cluster manager.
+func (cm *ClusterManager) Stop() {
 	klog.Info("Stopping the cluster manager")
-	metrics.UnregisterClusterManagerFunctional()
 	cm.defaultNetClusterController.Stop()
 	if config.OVNKubernetesFeature.EnableMultiNetwork {
 		cm.secondaryNetClusterManager.Stop()
 	}
-
-	cm.isActive = false
-	return nil
+	metrics.UnregisterClusterManagerFunctional()
```

Comment:
do we still need to unregister metrics now that its the same LE, so for CM as well we will exit when LE is lost right? I have a feeling we could make this similar to master and remove this altogether but @martinkennelly is the expert on this one.

Reply:
I added the unregister back because some HBO tests and the CM tests are starting the CM several times, and subsequent starts panic because metric registration fails.

I actually think it is technically correct to do even if it is not absolutely necessary in some scenarios.

But I will let @martinkennelly have the last word on this.

Reply:
> I added the unregister back because some HBO tests and the CM tests are starting the CM several times, and subsequent starts panic because metric registration fails.

ack, interesting..
> 
> I actually think it is technically correct to do even if it is not absolutely necessary in some scenarios.
> 
yes true, I agree as in unregistering something explicitly can't really cause harm I guess..sorry I made you go back/forth on this.